model/task_lifecycle.go

```diff
@@ -1393,7 +1396,40 @@ func doRestartFailedTasks(tasks []string, user string, results RestartResults) R
 	return results
 }
 
-func ClearAndResetStrandedTask(h *host.Host) error {
+// ClearAndResetStrandedContainerTask clears the container task dispatched to a
+// pod. It also resets the task so that the current task execution is marked as
+// finished and, if necessary, a new execution is created to restart the task.
+// TODO (PM-2618): should probably block single host task groups once they're
+// supported.
+func ClearAndResetStrandedContainerTask(p *pod.Pod) error {
+	runningTaskID := p.RunningTask
+	if runningTaskID == "" {
```

Comment:
I see some of the same patterns here as `ClearAndResetStrandedHostTask`. Might be worth some consolidation of the two functions and their logic. I'm not particularly opinionated on that though, up to you

Reply:
I'm not sure how to consolidate this without doing some weird muxing of host/pod docs together into a helper function. I think the loss of clarity by making a small empty-string-check helper that works for both struct types wouldn't be worth the consolidation.

Reply:
Ok, SGTM